contracts/LockDealBundleProvider/LockDealBundleProvider.sol

```diff
@@ -88,7 +87,12 @@ contract LockDealBundleProvider is
 
     function withdraw(
         uint256 poolId
-    ) public override onlyNFT onlyBundlePoolId(poolId) returns (uint256 withdrawnAmount, bool isFinal) {
+    ) public override onlyNFT returns (uint256 withdrawnAmount, bool isFinal) {
+        require(
+            poolIdToLockDealBundle[poolId].firstSubPoolId > 0,
+            "Pool is not a bundle pool"
+        );
```

Comment:
let's remove this. 
we got the `onlyNFT`
the NFT can call only the ID of the provider.